packages/core/logging/core-logging-server-internal/src/appenders/rolling_file/rolling_file_appender.ts

```diff
@@ -96,8 +116,9 @@ export class RollingFileAppender implements DisposableAppender {
     }
     this.isRolling = true;
     try {
-      await this.strategy.rollout();
+      await this.rollingStrategy.rollout();
       await this.fileManager.closeStream();
+      await this.retentionPolicy.apply();
```

Comment:
applying the retention policies occurs at the exact same place than when it was performed by the rolling strategy, so any potential error will be caught the exact same way, preserving the behavior.